packages/lesswrong/components/forumEvents/ForumEventPoll.tsx

```diff
@@ -405,147 +537,269 @@ export const ForumEventPoll = ({postId, hideViewResults, classes}: {
     if (currentUser) {
       const voteData: ForumEventVoteData = {
         forumEventId: event._id,
-        x: votePos,
-      }
+        x: newVotePos,
+      };
       if (!hasVoted) {
-        setVoteCount(count => count + 1)
-        setCurrentUserVote(votePos)
-        await addVote({variables: voteData})
-        refetch?.()
-        return
+        setVoteCount((count) => count + 1);
+        setCurrentUserVote(newVotePos);
+        await addVote({ variables: voteData });
+        refetch?.();
+        return;
       }
-      const delta = votePos - (currentUserVote ?? defaultVotePos)
+      const delta =
+        newVotePos - (currentUserVote ?? (DEFAULT_VOTE_INDEX / (NUM_TICKS - 1)));
       if (delta) {
-        voteData.delta = delta
-        setCurrentUserVote(votePos)
-        await addVote({variables: {
-          ...voteData,
-          ...(postId && {postIds: [postId]}),
-        }})
-        refetch?.()
+        voteData.delta = delta;
+        setCurrentUserVote(newVotePos);
+        await addVote({
+          variables: {
+            ...voteData,
+            ...(postId && { postIds: [postId] }),
+          },
+        });
+        refetch?.();
       }
-    }
-    // When a logged-out user tries to vote, just show the login modal
-    else {
+    } else {
       onSignup()
       void clearVote()
     }
   }, [
-    isDragging,
-    setIsDragging,
-    hasVoted,
+    currentBucketIndex,
     currentUser,
     addVote,
     event,
-    votePos,
     currentUserVote,
     postId,
     setCurrentUserVote,
     onSignup,
     clearVote,
-    refetch
-  ])
-  useEventListener("pointerup", saveVotePos)
+    refetch,
+    hasVoted,
+  ]);
+  useEventListener("pointerup", saveVotePos);
+
+  const { ForumIcon, LWTooltip, UsersProfileImage } = Components;
+
+  const ticks = Array.from({ length: NUM_TICKS }, (_, i) => i);
+
+  const tickPositions = useRef<number[]>([]);
+  const [votePos, setVotePos] = useState(
+    // Initial naive calculation which doesn't exactly line up with the ticks, so
+    // there isn't a large jump
+    (currentBucketIndex / (NUM_TICKS - 1)) * 100
+  );
+
+  useEffect(() => {
+    if (sliderRef.current) {
+      const sliderRect = sliderRef.current.getBoundingClientRect();
+      tickPositions.current = tickRefs.current.map((tick) => {
+        if (tick) {
+          const tickRect = tick.getBoundingClientRect();
+          return (
+            ((tickRect.left + (tickRect.width / 2) - sliderRect.left) /
+              sliderRect.width) *
+            100
+          );
+        }
+        return 0;
+      });
+      setVotePos(tickPositions.current[currentBucketIndex]);
+    }
+  }, [currentBucketIndex]);
+
+  useEffect(() => {
+    if (
+      tickPositions.current.length &&
+      tickPositions.current[currentBucketIndex] !== undefined
+    ) {
+      setVotePos(tickPositions.current[currentBucketIndex]);
+    }
+  }, [currentBucketIndex]);
```

Comment:
I've refactored this to remove the second `useEffect` and calculate the `votePos` inline:
```typescript
const votePos =
    tickPositions.current.length && tickPositions.current[currentBucketIndex] !== undefined
      ? tickPositions.current[currentBucketIndex]
      // Fall back to naive approximate calculation so there isn't a big jump after the first render
      : (currentBucketIndex / (NUM_TICKS - 1)) * 100;
```
(this also removes the need for `votePos` to be a `useState`)

The reason I ended up doing it this way is that I kept running into problems with aligning the result columns, ticks, and user vote, because there is padding and spacing that is complicated to account for exactly. This way the position of the user vote and the ticks should be more robust (e.g. if there is some kind of overflow that causes one tick to be wider then it should still line up). I'll add a comment to this effect.